src/stage1/migrate_info/balena_cfg_json.rs

```diff
@@ -230,3 +198,33 @@ impl BalenaCfgJson {
         &self.file
     }
 }
+
+fn check_api(api_endpoint: &str) -> Result<()> {
+    let res = Client::builder()
+        .build()
+        .upstream_with_context("Failed to create https client")?
+        .get(api_endpoint)
+        .send()
+        .upstream_with_context(&format!(
+            "Failed to send https request url: {}",
+            &api_endpoint
+        ))?;
+    debug!("Result = {:?}", res);
+    let status = res.status();
+    let response = res
+        .text()
+        .upstream_with_context("Failed to read response")?;
+
+    if status.is_success() && response.trim() == "OK" {
+        info!("connection to api: {} is ok", &api_endpoint);
+        Ok(())
+    } else {
+        return Err(Error::with_context(
+            ErrorKind::InvState,
+            &format!(
+                "failed to connect to api server @ {}. Your device might not come online",
+                &api_endpoint
+            ),
```

Comment:
If we reach this point we have already connected to the API server. The problem here is that it is unresponsive (or failing to reply to the ping, at least).

Suggestion (not tried on a compiler :-) ):

```suggestion
            &format!(
                "got an unexpected reply from the API server @ {}: '{}'",
                &api_endpoint,
                &response
            ),
```